mad-dog/src/tests/test_userReplicaSetNodes/index.js

```diff
@@ -0,0 +1,58 @@
+const { addAndUpgradeUsers } = require('../../helpers.js')
+const verifyValidCNs = require('./verifyValidCN')
+const deregisterCreatorNode = require('./deregisterCreatorNode.js')
+const stopRandomCreatorNode = require('./stopRandomCreatorNode.js')
+const setNumCreatorNodes = require('./setNumCreatorNodes.js')
+
+/**
+ * Tests that the user replica sets update when a node is deregistered or down
+ */
+const userReplicaSetNodes = async ({
+  numUsers = 10,
+  numCreatorNodes = 10,
+  iterations,
+  executeAll,
+  executeOne
+}) => {
+
+  // Make sure there are 4 creator nodes
+  let creatorNodeIDToInfoMapping = {}
+
+  let walletIndexToUserIdMap = {}
+  // Create users
+  // Initialize users
+  try {
+    walletIndexToUserIdMap = await addAndUpgradeUsers(
+      numUsers,
+      executeAll,
+      executeOne
+    )
+  } catch (e) {
+    return { error: `Issue with creating and upgrading users: ${e}` }
+  }
+
+
+  for (let iteration = 0; iteration < iterations; iteration++) {
+    // Ensure that there are 10 registered CN before each iteration
+    creatorNodeIDToInfoMapping = await setNumCreatorNodes(numCreatorNodes, executeOne)
+
+    const deregisteredCreatorNodeId = await deregisterCreatorNode(creatorNodeIDToInfoMapping)
+
+    await new Promise(resolve => setTimeout(resolve, 10 * 1000))
+    await verifyValidCNs(executeOne, executeAll, deregisteredCreatorNodeId, walletIndexToUserIdMap, creatorNodeIDToInfoMapping)
+
+    // Create a MadDog instance, responsible for taking down 1 node
+    const {
+      madDog,
+      removedCreatorNodeId
+    } = await stopRandomCreatorNode(creatorNodeIDToInfoMapping)
```

Comment:
would it be valuable to make sure that the previously deregisteredNode is not randomly selected in this step to be "stopped"?

Reply:
When the node is deregistered it is removed from the `creatorNodeIDToInfoMapping` variable, so it is not selected in this step. 
The `deregisterCreatorNode` function has a side effect of modifying the mapping.

Reply:
oh dope